reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -211,40 +317,47 @@ private synchronized void quickSlowestRS() {
             return;
         }
 
-        activeSockets.entrySet().forEach(e -> {
-            SocketAddress key = e.getKey();
-            WeightedSocket value = e.getValue();
-            logger.info("> " + key + " -> " + value);
+        activeSockets.forEach(value -> {
+            logger.info("> " + value);
         });
 
-        activeSockets.entrySet()
-            .stream()
-            .sorted((a,b) -> {
-                WeightedSocket socket1 = a.getValue();
-                WeightedSocket socket2 = b.getValue();
-                double load1 = 1.0/socket1.getPredictedLatency() * socket1.availability();
-                double load2 = 1.0/socket2.getPredictedLatency() * socket2.availability();
-                return Double.compare(load1, load2);
-            })
-            .limit(1)
-            .forEach(entry -> {
-                SocketAddress key = entry.getKey();
-                WeightedSocket slowest = entry.getValue();
-                try {
-                    logger.info("quicking slowest: " + key + " -> " + slowest);
-                    activeSockets.remove(key);
-                    slowest.close();
-                } catch (Exception e) {
-                    logger.warn("Exception while closing a ReactiveSocket", e);
+        WeightedSocket slowest = null;
+        double lowestAvailability = Double.MAX_VALUE;
+        for (WeightedSocket socket: activeSockets) {
+            double load = socket.availability();
+            if (socket.getPredictedLatency() != 0) {
+                load *= 1.0 / socket.getPredictedLatency();
+            }
+            if (load < lowestAvailability) {
+                lowestAvailability = load;
+                slowest = socket;
+                if (load == 0.0) {
```

Comment:
If the availability is `0.0` then all the above calculations are thrown away, so looks like this can be the first condition in this loop? Should we also handle negative availability?